tests/chain_sync_test.rs

```diff
@@ -81,10 +81,91 @@ mod chain_sync_test {
+        nigiri::try_cmd_repeatedly(nigiri::mine_blocks, 1, 10, HALF_SEC).unwrap();
+
+        sleep(Duration::from_secs(10));
+
+        assert_eq!(node.get_node_info().num_channels, 0);
+    }
+
+    fn start_node_open_confirm_channel_stop_node(node_handle: &NodeHandle) -> String {
+        let node = node_handle.start().unwrap();
+        let node_id = node.get_node_info().node_pubkey.to_hex();
+
+        let tx_id = nigiri::lnd_open_channel(&node_id).unwrap();
+
+        assert_eq!(node.get_node_info().num_channels, 1);
+        assert_eq!(node.get_node_info().num_usable_channels, 0);
+
+        nigiri::try_cmd_repeatedly(nigiri::mine_blocks, 6, 10, HALF_SEC).unwrap();
+
+        sleep(Duration::from_secs(10));
+
+        assert_eq!(node.get_node_info().num_channels, 1);
+        assert_eq!(node.get_node_info().num_usable_channels, 1);
+
+        tx_id
+    }
+
+    fn start_node_open_channel_without_confirm_stop_node(node_handle: &NodeHandle) -> String {
+        let node = node_handle.start().unwrap();
+        let node_id = node.get_node_info().node_pubkey.to_hex();
+
+        let tx_id = nigiri::lnd_open_channel(&node_id).unwrap();
+
+        assert_eq!(node.get_node_info().num_channels, 1);
+        assert_eq!(node.get_node_info().num_usable_channels, 0);
+
+        tx_id
+    }
+
+    #[test]
+    fn test_channel_remains_usable_after_restart() {
+        let node_handle = setup();
+
+        start_node_open_confirm_channel_stop_node(&node_handle);
+
+        let node = node_handle.start().unwrap();
+
+        sleep(Duration::from_secs(10));
+
+        assert_eq!(node.get_node_info().num_channels, 1);
+        assert_eq!(node.get_node_info().num_usable_channels, 1);
+    }
+
+    #[test]
+    fn test_channel_is_confirmed_chain_only_after_6_confirmations_offline_node() {
+        let node_handle = setup();
+
+        start_node_open_channel_without_confirm_stop_node(&node_handle);
+
+        nigiri::try_cmd_repeatedly(nigiri::mine_blocks, 6, 10, HALF_SEC).unwrap();
+        sleep(Duration::from_secs(5));
+
+        let node = node_handle.start().unwrap();
+
+        assert_eq!(node.get_node_info().num_channels, 1);
+        assert_eq!(node.get_node_info().num_usable_channels, 1);
+    }
+
+    #[test]
+    fn test_force_close_is_detected_offline_node() {
+        let node_handle = setup();
+
+        let tx_id = start_node_open_confirm_channel_stop_node(&node_handle);
+
+        nigiri::lnd_force_close_channel(tx_id).unwrap();
+        // TODO: as soon as we regularly reconnect to peers, we can uncomment the following line
+        //      as then we'll be able to handle not being connected to our peers
+        // nigiri::lnd_stop().unwrap();
+
+        nigiri::try_cmd_repeatedly(nigiri::mine_blocks, 1, 10, HALF_SEC).unwrap();
+
+        let node = node_handle.start().unwrap();
 
         sleep(Duration::from_secs(10));
 
+        // This only passes with the sleep that precedes it. TODO: confirm that's not a problem
```

Comment:
Is this comment really only relevant for this test case? The other's don't have that comment. Is it still required? Is the TODO still open?

Reply:
It's mainly relevant here. In other situations is normal that we need to wait > 5 secs because we sync every 5 seconds. We wait 10 secs to be sure the node syncs. In this case, I'm not sure why the wait is needed because we sync on start.